shipyard/src/test/java/org/jclouds/shipyard/features/RolesApiLiveTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.shipyard.features;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertNotNull;
+
+import org.jclouds.shipyard.domain.roles.RoleInfo;
+import org.jclouds.shipyard.internal.BaseShipyardApiLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "RolesApiLiveTest", singleThreaded = true)
+public class RolesApiLiveTest extends BaseShipyardApiLiveTest {
+
+   private final String roleName = "jclouds-shipyard-test-" + UUID.randomUUID().toString().replaceAll("-", "");
+   
+   @AfterClass (alwaysRun = true)
+   protected void tearDown() {
+      boolean removed = api().deleteRole(roleName);
+      assertTrue(removed);
+   }
+   
+   public void testCreateRole() throws Exception {
+     api().createRole(roleName);
+   }
+   
+   @Test (dependsOnMethods = "testCreateRole")
+   public void testGetRole() throws Exception {
+      RoleInfo role = api().getRole(roleName);
+      assertNotNull(role, "Role was not set");
+      assertTrue(role.name().equals(roleName), "Found Role name does not match expected name: found=" + role.name() + ", expected=" + roleName);
+   }
+   
+   @Test (dependsOnMethods = "testGetRole")
+   public void testListRoles() throws Exception {
+      List<RoleInfo> possibleRoles = api().listRoles();
+      assertNotNull(possibleRoles, "possibleRoles was not set");
+      assertTrue(possibleRoles.size() > 0, "Expected at least 1 Role but list was empty");
+      boolean roleFound = false;
+      for (RoleInfo possibleRole : possibleRoles) {
+         if (possibleRole.name().equals(roleName)) {
+            roleFound = true;
+         }
+      }
```

Comment:
WTF?!? It appears that in my hazy cloud of sickness I'm currently entertaining that I made the changes to the ServiceKeys API instead of the Roles API. Still needed to be done there so that's a plus, but my apologies for seemingly completely ignoring all your comments that I thought I was addressing. I'll hop on those now ;)